protos/flyteidl/core/tasks.proto

```diff
@@ -154,4 +154,44 @@ message Container {
     // Ports to open in the container. This feature is not supported by all execution engines. (e.g. supported on K8s but
     // not supported on AWS Batch)
     repeated ContainerPort ports = 7;
+
+    // BETA: This enables use of CoPilot. This makes it possible to to run a completely portable container, that uses inputs and outputs
+    // only from the local file-system and without having any reference to flyteidl. This is supported only on K8s at the moment.
+    // If CoPilot is enabled, then data will be mounted in accompanying directories specified in the CoPilot settings. If the directories 
+    // are not specified, inputs will be mounted onto and outputs will be uploaded from a pre-determined file-system path. Refer to the documentation
+    // to understand the default paths.
+    bool use_copilot = 8;
+
+    // Optional configuration for CoPilot. If not specified, then default values are used.
+    CoPilot copilot_config = 9;
+}
+
+// This configuration allows executing raw containers in Flyte using the Flyte CoPilot system.
+// Flyte CoPilot, eliminates the needs of flytekit or sdk inside the container. Any inputs required by the users container are side-loaded in the input_path
+// Any outputs generated by the user container - within output_path are automatically uploaded.
+message CoPilot {
```

Comment:
Should we call it `MetadataConfig` or `InputOutputMetadataConfig` (too long)? just something descriptive you don't need to read the docs to understand kind of thing

Reply:
DataConfig?

Reply:
renamed to DataLoadingConfig